src/math/ai/statemachine.ts

```diff
@@ -0,0 +1,125 @@
+type EnterCallback = (sm: StateMachine, previous: string) => void;
+type UpdateCallback = (sm: StateMachine, current: string) => void;
+type DrawCallback = (sm: StateMachine, current: string) => void;
+type ExitCallback = (sm: StateMachine, next: string) => void;
+type TransitionCallback = (sm: StateMachine, from: string, to: string) => void;
+
+type StateCallbacks = {
+    enter?: EnterCallback;
+    update?: UpdateCallback;
+    draw?: DrawCallback;
+    exit?: ExitCallback;
+};
+
```

Comment:
There's a reason to use Type Aliases instead of just put the type? I think it makes documentation verbose

Reply:
Our documentation process export the types even if they're not "exported" because it checks for the references, so they will appear anyway

Reply:
So how can we make sure the docs don't pick up internal types, functions and classes?

Reply:
But they have to, if no how an user would know what have to pass here in `cb`:

https://github.com/kaplayjs/kaplay/blob/1fd31ea36ed35b2b395329cad434474373d45173/src/math/ai/statemachine.ts#L70-L75

I mean, then the user must have to click `EnterCallback` to know what to pass here, imagine we did this with all our functions that would be a lot of types that you have to click on for knowing them